src/plugins/preauth/pkinit/pkinit_crypto_openssl.c

```diff
@@ -3557,24 +3557,32 @@ pkinit_C_LoadModule(const char *modname, CK_FUNCTION_LIST_PTR_PTR p11p)
     long err;
     CK_RV rv;
 
-    pkiDebug("loading module \"%s\"... ", modname);
+    TRACE_PKINIT_CLIENT_PKCS11_OPEN(context, modname);
     if (krb5int_open_plugin(modname, &handle, &einfo) != 0) {
-        pkiDebug("not found\n");
+        TRACE_PKINIT_CLIENT_PKCS11_OPEN_FAILED(context, einfo.msg ?
+                                               einfo.msg : "Unknown error");
+        k5_clear_error(&einfo);
```

Comment:
This is I think correct, but should be called out in the commit message, or possibly in a separate commit.

Reply:
I am unclear exactly what you mean here; do you want the commit message to explicitly mention that we log the error from loading the PKCS#11 module?  I can do that, but I am just trying to understand exactly what you want.

(The rest of the suggestions look fine and I will implement them)

Reply:
Apologies, that wasn't very clear.

I think you fixed a memory leak here by adjusting where `k5_clear_error` is called.